source/fundamentals/linq.txt

```diff
@@ -0,0 +1,541 @@
+documents from the "joined" collection.
+
+.. note:: 
+
+   To perform a lookup, you must make both collections queryable using the
+   ``AsQueryable`` method.
+
+   To learn how to make a collection queryable, see :ref:`csharp-linq-queryable`.
+
+Consider a second collection in the ``sample_restaurants`` database called
+``reviews`` that has restaurant reviews. You can join documents from that collection
+to documents with the same ``name`` value in the ``restaurants`` collection using
+the ``$lookup`` stage.
+
+The following ``Review`` class models the documents in the ``reviews`` collection:
+
+.. literalinclude:: /includes/fundamentals/code-examples/linq.cs
+   :language: csharp
+   :dedent:
+   :start-after: start-review-model
+   :end-before: end-review-model
+
+Select the :guilabel:`Query Syntax` or :guilabel:`Method Syntax` tab to see how 
+to generate a ``$lookup`` stage using LINQ:
+
+.. tabs::
+
+   .. tab:: Query Syntax
+      :tabid: query-syntax
+
+      .. code-block:: csharp
+
+         var query = from restaurant in queryableCollection
+             join rv in reviewCollection on restaurant.Name equals rv.RestaurantName into reviews
+             select new { restaurant, reviews };
+
+   .. tab:: Method Syntax
+      :tabid: method-syntax
+
+      .. code-block:: csharp
+
+         var query = queryableCollection
+             .GroupJoin(reviewCollection,
+                 restaurant => restaurant.Name,
+                 review => review.RestaurantName,
+                 (restaurant, reviews) =>
+                     new {restaurant, reviews}
+             );
+
+The preceding example returns all documents from the ``restaurants`` collection. Each
+restaurant document has an added field called ``reviews``, which contains all 
+reviews for that restaurant. A review matches a restaurant if the value of the
+``name`` field in the review document matches the ``name`` field of the restaurant
+document.
+
+The following is a subset of the returned results:
+
+.. code-block:: json
+
+   // Results Truncated
+
+   { "restaurant" : {
+      "_id" : ObjectId("..."),
+      "name" : "The Movable Feast", 
+      "restaurant_id" : "40361606", 
+      "cuisine" : "American", 
+      "address" : {...}, 
+      "borough" : "Brooklyn", 
+      "grades" : [...] }, 
+      "reviews" : [
+        { "_id" : ObjectId(...), "restaurant_name" : "The Movable Feast", "reviewer" : "Lazlo Cravensworth", "review_text" : "Great restaurant! 12/10 stars!" },
+        { "_id" : ObjectId("..."), "restaurant_name" : "The Movable Feast", "reviewer" : "Michael Scarn", "review_text" : "It really was a feast" }
+       ]
+    }
+
+Unsupported Aggregation Stages
+------------------------------
+
+The {+driver-long+} implementation of LINQ does not support the following
+aggregation stages:
+
+- ``$redact``
+- ``$geoNear``
+- ``$out``
+
+Supported Methods
+-----------------
+
+The {+driver-long+} implementation of LINQ supports the following methods:
```

Comment:
I think this is not a full list of ALL supported methods.

This just looks like a list of supported "terminator methods" (not sure if that's an official term).

A terminator method is one that immediately executes the `IQueryable` it is applied to and then returns a scalar result.

Reply:
Is there a list of all supported methods that I could link to here for coverage? Or would it be better to change the header here to say something along the lines of "These are some of the supported methods"?

Reply:
Probably easiest for now to just just say "some".

We don't have a complete list of supported methods and the list is constantly growing.